File: java_solutions/3024.type-of-triangle.java
/*
 * @lc app=leetcode id=3024 lang=java
 *
 * [3024] Type of Triangle
 */

class Solution {
    public String triangleType(int[] nums) {
        return Arrays.stream(nums).sorted().toArray() instanceof int[] s && s[0] + s[1] <= s[2] ? "none" : 
            Arrays.stream(nums).distinct().count() == 1 ? "equilateral" : 
            Arrays.stream(nums).distinct().count() == 2 ? "isosceles" : "scalene";
    }
}

File: kotlin_solutions/3024.type-of-triangle.kt
/*
 * @lc app=leetcode id=3024 lang=kotlin
 *
 * [3024] Type of Triangle
 */

class Solution {fun triangleType(nums: IntArray) = nums.sorted().let { sorted -> if (sorted[0] + sorted[1] <= sorted[2]) "none" else when (nums.toSet().size) { 1 -> {"equilateral"} 2 -> {"isosceles"} else -> "scalene" } }}
